blosc/schunk.c

```diff
@@ -314,7 +314,37 @@ int blosc2_schunk_append_chunk(blosc2_schunk *schunk, uint8_t *chunk, bool copy)
   /* Update counters */
   schunk->nchunks = nchunks + 1;
   schunk->nbytes += nbytes;
-  schunk->cbytes += cbytes;
+  if (schunk->frame == NULL) {
+    schunk->cbytes += cbytes;
+  }
+  else {
+    // A frame
+    bool all_zeros = chunk[BLOSC2_CHUNK_BLOSC2_FLAGS] & (BLOSC2_ZERO_RUNLEN << 4);
+    bool all_nans = chunk[BLOSC2_CHUNK_BLOSC2_FLAGS] & (BLOSC2_NAN_RUNLEN << 4);
+    bool all_repeats = all_nans & all_zeros;
+    if (all_repeats) {
+      // A non-special chunk is always stored in frames
+      schunk->cbytes += cbytes;
+    }
+    else if (all_zeros || all_nans) {
+      if (schunk->cbytes == 0) {
+        // Increase by 0 just to be explicit
+        schunk->cbytes += 0;
+      }
+    }
+    else {
+      // A non-special chunk is always stored in frames
+      schunk->cbytes += cbytes;
+    }
```

Comment:
This can be simplified:
```
if (all_zeros || all_nans) {
  schunk->cbytes += 0;
}
else {
  schunk->cbytes += cbytes;
}
```

Reply:
Not possible because `all_repeats` includes `all_zeros` and `all_nans` (both bits are set at the same time).